### PR TITLE
Refactor tasklist

### PR DESCRIPTION
**Pull request by sealDebolina**
Refactoring FL401 applicant and C100 respondent to common tasklist.

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-4374
https://tools.hmcts.net/jira/browse/PRL-4373

### Change description ###

"As part of this technical refactoring task, use the task-list manage module I have created and use the config to manage DA applicant and CA respondent task-list landing page."

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by vivek-sekhar — APPROVED** [review]

File: src/main/routes.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import { StepWithContent, stepsWithContent } from './steps/';
 import { AccessibilityStatementGetController } from './steps/accessibility-statement/get';
 import ApplicantConfirmContactDetailsPostController from './steps/applicant/confirm-contact-details/checkanswers/controller/ApplicantConfirmContactDetailsPostController';
 import { SupportYouNeedDuringYourCaseController } from './steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController';
-import ApplicantTaskListGetController from './steps/applicant/task-list/get';
+//import ApplicantTaskListGetController from './steps/applicant/task-list/get';
```
Comment (AlokDatta) [inline src/main/routes.ts:21]: Do we need commented out code?

File: src/main/routes.ts [hunk 2]
```diff
@@ -153,6 +153,7 @@ export class Routes {
     app.get(ACCESSIBILITY_STATEMENT, errorHandler(new AccessibilityStatementGetController().get));
     app.get(CONTACT_US, errorHandler(new ContactUsGetController().get));
     app.get(`${APPLICANT_TASK_LIST_URL}/:caseId`, errorHandler(new GetCaseController().fetchAndRedirectToTasklist));
+    //app.get(`${APPLICANT_TASK_LIST_URL}/:caseId`, errorHandler(new CaseDetailsGetController().get));
```
Comment (AlokDatta) [inline src/main/routes.ts:156]: Do we need commented out code?

File: src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.test.ts [hunk 3]
```diff
@@ -213,4 +213,27 @@ describe('SupportYouNeedDuringYourCaseController', () => {
     await controller.post(req, res);
     expect(req.session.userCase.applicantsFL401.response.keepDetailsPrivate).toEqual(undefined);
   });
+
+  test('Should update the SupportYouNeedDuringYourCase details if user id doesnot matches with respondent for DA', async () => {
+    req.session.user.id = '0c09b130-2eba-4ca8-a910-1f001bac01e7';
```
Comment (AlokDatta) [inline src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.test.ts:218]: is this a dummy user id? if yes, please ignore this comment and mark it as resolved.

**Review by AlokDatta — APPROVED** [review]
very minor review comments. merge it once addressed. Sonar looks good.

File: src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.test.ts (hunk 3 above)
Comment (vivek-sekhar) [reply src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.test.ts:218]: Yes it is a mock user ID.

File: src/main/routes.ts (hunk 1 above)
Comment (hcarslaw) [reply src/main/routes.ts]: Removed

File: src/main/routes.ts (hunk 2 above)
Comment (hcarslaw) [reply src/main/routes.ts]: Removed

**Pull request merged**